plugin/gitgutter.vim

```diff
@@ -207,6 +207,10 @@ augroup gitgutter
   autocmd CursorHold,CursorHoldI * call gitgutter#process_buffer(bufnr(''), 0)
   autocmd FileChangedShellPost   * call gitgutter#process_buffer(bufnr(''), 1)
 
+  " Reset cached filepath when e.g. executing :saveas on a tracked file
+  " autocmd BufWritePost * call s:on_bufwritepost()
+  autocmd BufWritePost * let b:gitgutter.path = expand('%:p')
```

Comment:
This will only work if the current directory is the project root directory, i.e. not all the time.

Reply:
You could try `unlet b:gitgutter.path`.  (Don't have time to test it right now...)